tests/virt_autotest/kubevirt_tests_server.pm

```diff
@@ -0,0 +1,372 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: This kubevirt test relies on the upstream test code, which is downstreamed as virt-tests.
+#          This is the part running on server node.
+# Maintainer: Nan Zhang <nan.zhang@suse.com>
+
+use base multi_machine_job_base;
+use strict;
+use warnings;
+use testapi;
+use lockapi;
+use mmapi;
+use utils;
+use File::Basename;
+
+my @kubevirt_tests = (
+    'vmi_lifecycle_test',
+    'vm_test',
+    'vmipreset_test',
+    # 'vmi_configuration_test', # Disable this test, because VMI Huge page settings is required
+    'vmi_headless_test',
+    'access_test',
+    'storage_test',
+    'container_disk_test',
+    'datavolume_test',
+    'hotplug_test',
+    'vmi_cloudinit_test',
+    'vmi_iothreads_test',
+    'vmi_multiqueue_test',
+    'primary_pod_network_test',
+    'vmi_multus_test',
+    'vmi_networking_test',
+    'networkpolicy_test',
+    'vmi_monitoring_test',
+    'vnc_test',
+    'console_test',
+    'credentials_test',
+    'expose_test',
+    'replicaset_test',
+    'network_services_test',
+    'vmi_kernel_boot_test',
+    'infra_test',
+    'operator_test',
+    'migration_test',
+    'subresource_api_test',
+    'virt_control_plane_test'
+);
+
+sub run {
+    my ($self) = shift;
+    my $sut_ip = get_required_var('SUT_IP');
+
+    set_var('SERVER_IP', $sut_ip);
+    bmwqemu::save_vars();
+
+    # Synchronize the server & agent nodes
+    # mutex_wait('KUBEVIRT_TEST_SETUP');
+    barrier_create('KUBEVIRT_TEST_SETUP', 2);
+    barrier_wait('KUBEVIRT_TEST_SETUP');
```

Comment:
It seems you create and wait on the same signal on both agent and server ? Not only KUBEVIRT_TEST_SETUP, but also many others.

Reply:
Refer to the latest discussion https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/14364#discussion_r896533428